make.bat

```diff
@@ -98,64 +208,62 @@ if exist "%InstallDir%\Common7\Tools\vsdevcmd.bat" (
 
 set ObjFile=.v.c.obj
 
-cl.exe /volatile:ms /Fo%ObjFile% /O2 /MD /D_VBOOTSTRAP vc\v_win.c user32.lib kernel32.lib advapi32.lib shell32.lib /link /nologo /out:v.exe /incremental:no>>%log_file% 2>>&1
+if !verbose_log! EQU 1 ( echo [Compile] cl.exe /volatile:ms /Fo%ObjFile% /O2 /MD /D_VBOOTSTRAP vc\v_win.c user32.lib kernel32.lib advapi32.lib shell32.lib /link /nologo /out:v.exe /incremental:no>>!log_file! )
+cl.exe /volatile:ms /Fo%ObjFile% /O2 /MD /D_VBOOTSTRAP vc\v_win.c user32.lib kernel32.lib advapi32.lib shell32.lib /link /nologo /out:v.exe /incremental:no>>!log_file! 2>>&1
 if %ERRORLEVEL% NEQ 0 goto :compile_error
 
 echo  ^> Compiling with .\v.exe self
-v.exe -cc msvc self>>%log_file% 2>>&1
-del %ObjFile%>>%log_file% 2>>&1
+if !verbose_log! EQU 1 ( echo [Make] v.exe -cc msvc self>>!log_file! )
+v.exe -cc msvc self>>!log_file! 2>>&1
+del %ObjFile%>>!log_file! 2>>&1
 if %ERRORLEVEL% NEQ 0 goto :compile_error
 goto :success
 
-:fresh_tcc
-rd /s /q "%tcc_dir%"
-
-:clone_tcc
-git clone --depth 1 --quiet "%tcc_url%" "%tcc_dir%"
-set cloned_tcc=1
-goto :tcc_strap
-
 :tcc_strap
 echo.
 echo Attempting to build v.c with TCC...
 
 where /q tcc
 if %ERRORLEVEL% NEQ 0 (
-	if exist "%tcc_dir%" (
-		set tcc_exe="%tcc_dir%\tcc.exe"
-	) else if "%cloned_tcc%"=="" (
-		echo  ^> TCC not found
-		echo  ^> Downloading TCC from %tcc_url%
-		goto :clone_tcc
-	) else (
-		echo  ^> TCC not found, even after cloning %cloned_tcc%
-		goto :error
-	)
+	if !compiler_opt! EQU "fresh-tcc" (
+        rd /s /q "%tcc_dir%">NUL 2>&1
+        set /a valid_cc=1
+    ) else if !compiler_opt! EQU "tcc" ( set /a valid_cc=1 )
+    if not exist "%tcc_dir%" (
+        echo  ^> TCC not found
+        echo  ^> Downloading TCC from %tcc_url%
+        git clone --depth 1 --quiet "%tcc_url%" "%tcc_dir%"
+    )
+    pushd %tcc_dir% || (
+        echo  ^> TCC not found, even after cloning
+        goto :error
+        exit /b 1
```

Comment:
why is an exit needed after the `goto :error` line ?

Reply:
right, initially I thought for some reason `goto ;error` was `call :error`, I will fix this